crates/walrus-service/src/node/config.rs

```diff
@@ -788,6 +792,93 @@ impl GarbageCollectionConfig {
     }
 }
 
+/// Entry defining a size bucket and the corresponding deferral duration.
+#[serde_as]
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq)]
+pub struct SizeDeferralEntry {
+    /// Maximum unencoded blob size (inclusive) in bytes for this bucket.
+    pub max_unencoded_bytes: u64,
+    /// Deferral duration to apply for this bucket.
+    #[serde_as(as = "DurationSeconds<u64>")]
+    #[serde(rename = "defer_secs")]
+    pub defer: Duration,
+}
+
+/// Configuration that controls deferring recovery when a live client upload is likely.
+#[serde_as]
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq)]
+#[serde(default)]
+pub struct LiveUploadDeferralConfig {
+    /// Enables applying a deferral window based on blob size.
+    pub enabled: bool,
+    /// Lookup table from size upper-bounds to deferral durations. Earlier entries take precedence.
+    #[serde(alias = "table")]
+    pub buckets: Vec<SizeDeferralEntry>,
```

Comment:
Suggestion: document that when this is empty, max_total_defer is applied to all sizes.

Reply:
done